io/src/main/scala/sbt/io/Hash.scala

```diff
@@ -52,7 +52,18 @@ object Hash {
   def apply(as: Array[Byte]): Array[Byte] = apply(new ByteArrayInputStream(as))
 
   /** Calculates the SHA-1 hash of the given file.*/
-  def apply(file: File): Array[Byte] = Using.fileInputStream(file)(apply)
+  def apply(file: File): Array[Byte] = {
+    val key = (file, file.length, file.lastModified)
```

Comment:
I think `file.length` is redundant as a key component. If it changes, `file.lastModified` should logically also change, always. Hence we can use `file -> file.lastModified` as the lookup key with the same properties.

Reply:
if we use the NIO API you basically get both for free... so why not.